src/player/player.ts

```diff
@@ -714,5 +716,33 @@ module Shumway.Player {
       });
       symbol.resolveAssetPromise.then(symbol.resolveAssetCallback, null);
     }
+
+    private _whitelist: {protocol: string; hostname: string; insecure: boolean}[] = [];
+    private _allowInsecure: boolean = false;
+
+    addToWhitelist(domain: string, insecure: boolean) {
+      if (domain.indexOf('://') < 0) { // this includes '*'
+        this._whitelist.push({protocol: 'http:', hostname: domain, insecure: insecure});
+        return;
+      }
+      try {
+        var url = new (<any>window).URL(domain);
+        this._whitelist.push({protocol: url.protocol, hostname: url.hostname, insecure: insecure});
+      } catch (e) { }
+    }
+
+    checkDomain(domain: string): boolean {
+      try {
+        var url = new (<any>window).URL(domain);
+      } catch (e) {
+        return false;
+      }
+      return this._whitelist.some(function (entry) {
+        if (url.hostname !== entry.hostname && entry.hostname !== '*') {
+          return false;
+        }
+        return entry.insecure || (url.protocol === 'https:' || entry.protocol !== 'https:');
```

Comment:
can we just check `entry.insecurre || (url.protocol === entry.protocol)`, or is that functionally different?